docs/source/nodes_and_pipelines/pipeline_registry.md

```diff
@@ -0,0 +1,73 @@
+
+```python
+def register_pipelines() -> Dict[str, Pipeline]:
+    """Register the project's pipelines.
+
+    Returns:
+        A mapping from pipeline names to ``Pipeline`` objects.
+    """
+    data_processing_pipeline = dp.create_pipeline()
+    data_science_pipeline = ds.create_pipeline()
+
+    return {
+        "__default__": data_processing_pipeline + data_science_pipeline,
+        "data_processing": data_processing_pipeline,
+        "data_science": data_science_pipeline,
+    }
+```
+
+```{note}
+The order in which you add the pipelines together is not significant (`data_science_pipeline + data_processing_pipeline` would produce the same result), since Kedro automatically detects the data-centric execution order for all the nodes in the resulting pipeline.
+```
+
+## Pipeline autodiscovery
+
+In the above example, you need to update the `register_pipelines()` function whenever you create a pipeline that should be returned as part of the project's pipelines. Since Kedro 0.18.3, you can achieve the same result with less code using `find_pipelines()`[/kedro.framework.project.find_pipelines]:
+
+```python
+def register_pipelines() -> Dict[str, Pipeline]:
+    """Register the project's pipelines.
+
+    Since Kedro 0.18.3, projects can use the ``find_pipelines`` function
+    to autodiscover pipelines. However, projects that require more fine-
+    grained control can still construct the pipeline mapping without it.
+
+    Returns:
+        A mapping from pipeline names to ``Pipeline`` objects.
+    """
+    pipelines = find_pipelines()
+    pipelines["__default__"] = sum(pipelines.values())
+    return pipelines
+```
+
+Under the hood, the `find_pipelines()` function traverses the `src/<package_name>/pipelines/` directory, returning a mapping from pipeline directory name to [`Pipeline`](/kedro.pipeline.Pipeline) object by:
+
+1. importing the `<package_name>.pipelines.<pipeline_name>` module
+1. calling the `create_pipeline()` function exposed by the `<package_name>.pipelines.<pipeline_name>` module
+1. validating that the constructed object is a [`Pipeline`](/kedro.pipeline.Pipeline)
+
+If any of these steps fail, `find_pipelines()` raises an appropriate warning and skips the current pipeline but continues traversal.
+
+The mapping returned by `find_pipelines()` can be modified. For example, to add a data engineering pipeline that isn't part of the default pipeline, add it to the dictionary *after* constructing the default pipeline:
```

Comment:
Just for the record, this is something very common for me when I use namespaces or modular pipelines: I duplicate and modify slightly the pipeline at this moment. 

Combining pipelines as described below is quite common too, sometimes I want to be able to run two pipelines separately (e.g. de, ds) or at the same time (de+ds).

Reply:
I think it would be nice to have another example here, or maybe just a note to suggest some other things users might like to do:
* filter pipelines like `pipelines["data_processing"] = pipelines["data_processing"].filter(tags=..., from_nodes=...)`
* put this before the definition of `__default__` if you want it reflected when you do `kedro run` without specifying a pipeline

Also I think we should have a sentence that explains what that `pipelines["__default__"]` does. Maybe after you say "The mapping returned by `find_pipelines()` can be modified.", something like "The default template also defines a `__default__` pipeline (executed by `kedro run` when blah blah). This is the concatenation of all blah blah blah".

Reply:
How about https://github.com/kedro-org/kedro/commit/4db786b5d2316fb32e73df420d3b1fc006944970? @AntonyMilneQB I didn't find it necessary to add examples of `filter`, `from_nodes`, etc.; I think this is sufficiently documented elsewhere, and isn't exactly pertinent. I did go ahead and re-explain the default pipeline, providing a link (even though one could argue that need not be done here).

Reply:
@deepyaman This looks good. The main thing I thought might require additional explanation though is what `sum(pipelines.values())` does, since this is quite new kedro syntax and maybe not obvious to a beginner user. Something about how it concatenates together all the nodes from pipelines in the `pipelines` dictionary. 

But this is not a big thing, feel free to leave out if you think it's not useful.